### PR TITLE
#9: refactor for tests and add a first test (closes #9)

### LINKED ISSUE #9: refactor for tests and add a first test
## requirements

- currently this app is difficult to test
- refactor the express app to separate routes from the server, and write a first test of a route
  - test that `/slackCommand` returns an error if the command is not formatted correctly

## specs

- use `express.Router` to separate routing logic from http server
- import the router and test against that (do not test via network)

### PR DESCRIPTION
**Pull request by lucacioria**
Issue #9

### tests performed
- informally tested with `/pledge2` but not covered much (just basic pledge and list, no notifications)
- added unit tests.. idea is that I'll add more tests soon so I didn't bother to re-test everything now

**I took a backup ;-)**

### tests not performed (domain coverage)
most of them.. :(

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import 'babel-core/register';
+import 'babel-polyfill';
+import express from 'express';
+import router from './routes';
+
+const app = express();
+app.use('/', router);
+
+module.exports = app;
```
Comment (FrancescoCioria) [inline src/app.js]: in ES6 this should be `export default app`

File: src/routes.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    // notify on slack
+    const notificationMessage = `pledge "${content}" has been completed !!!`;
+    await postOnSlack({
+      text: notificationMessage,
+      channel: performer
+    });
+    await postOnSlack({
+      text: notificationMessage,
+      channel: requester
+    });
```
Comment (gabro) [inline src/routes.js:122]: You can execute these two in parallel by instantiating both promises and then waiting.
E.g.

val p1 = somePromise
val p2 = someOtherPromise
// both promises are running now
await p1
await p2

or even

await Promise.all(p1, p2)

File: src/routes.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import 'babel-core/register';
```
Comment (FrancescoCioria) [inline src/routes.js]: @gabro usually imports like this one live better in the entry file (aka `index.js`). Why did you leave them here?

File: tests/index.test.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          expect(typeof res.text).toBe('string');
+          // headers for lists are presents
+          expect(res.text).toMatch('My pledges');
+          expect(res.text).toMatch('My requests');
+          // no pledge is present (empty lists)
+          expect(res.text).not.toMatch('pledged to');
+        });
+    });
+
+    it.only('notifies both requester and performer when a pledge is created', () => {
```
Comment (FrancescoCioria) [inline tests/index.test.js]: what's `it.only`? is it some kind of `fit`?

File: src/routes.js (hunk 3 above)
Comment (lucacioria) [reply src/routes.js]: @FrancescoCioria consider that `index.js` is used only in production (`yarn start`) but in testing (`yarn test`) I start from `app.js` (`index.js` is not imported at all).

File: tests/index.test.js (hunk 4 above)
Comment (lucacioria) [reply tests/index.test.js]: oooopss, it should not be committed, it only runs that test! I'll fix

File: src/routes.js (hunk 3 above)
Comment (gabro) [reply src/routes.js]: that import is useless. I thought I already removed them all, I'm going to purge this one two.

Comment (gabro) [reply src/routes.js]: done

File: src/routes.js (hunk 2 above)
Comment (gabro) [reply src/routes.js:122]: @lucacioria ^

**Pull request merged**